frontend/src/components/AddClassSearchModal.tsx

```diff
@@ -0,0 +1,308 @@
+
+interface AddClassSearchModalProps {
+  visible: boolean;
+  handleClose: () => void;
+  handleSubmit: (courses: ScheduleCourse[]) => void;
+}
+
+const CloseButtonWrapper = styled.div`
+  position: absolute;
+  top: 12px;
+  right: 18px;
+`;
+
+const OuterSection = styled.div`
+  position: fixed;
+  background: white;
+  width: 35%;
+  height: auto;
+  top: 40%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  outline: none;
+  padding-left: 35px;
+  padding-bottom: 25px;
+  padding-right: 35px;
+`;
+
+const InnerContainer = styled.div`
+  width: 100%;
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const SearchContainer = styled.div`
+  width: 100%;
+`;
+
+const SearchResultsScrollContainer = styled.div`
+  height: 200px;
+  width: 100%;
+  margin-top: 20px;
+  margin-bottom: 10px;
+  border: 1px solid red;
+  border-radius: 10px;
+  overflow-y: scroll;
+  font-family: Roboto;
+  font-style: normal;
+  font-weight: normal;
+`;
+
+const SearchResultContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  padding: 10px;
+  margin-left: 15px;
+  margin-right: 30px;
+`;
+
+const NoResultContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  padding: 10px;
+  margin-left: 15px;
+  margin-right: 30px;
+`;
+
+const Loading = styled.div`
+  font-size: 10px;
+  line-height: 10px;
+  margin-top: 10px;
+  margin-bottom: 5px;
+  margin-left: 10px;
+  margin-right: 10px;
+`;
+
+const AddedClassesContainer = styled.div``;
+
+const ResultInfoContainer = styled.div``;
+
+const AddClassButton = styled.div`
+  padding: 5px;
+  background: pink;
+  width: 18px;
+  height: 18px;
+  -moz-border-radius: 50px;
+  -webkit-border-radius: 50px;
+  border-radius: 50px;
+  align-items: center;
+  &:hover {
+    background: red;
+  }
+`;
+
+const AddClassError = styled.div`
+  background: pink;
+  color: white;
+  width: 18px;
+  height: 18px;
+  padding: 5px;
+  font-size: 16px;
+  font-weight: 800;
+  -moz-border-radius: 50px;
+  -webkit-border-radius: 50px;
+  border-radius: 50px;
+  text-align: center;
+`;
+
+const SubjectId = styled.div`
+  font-size: 10px;
+  line-height: 21px;
+`;
+
+const CourseName = styled.div`
+  font-size: 15px;
+  line-height: 21px;
+`;
+
+const EMPTY_COURSE_LIST: ScheduleCourse[] = [];
+
+interface SearchResultProps {
+  course: ScheduleCourse;
+}
+
+export const AddClassSearchModal: React.FC<
+  AddClassSearchModalProps
+> = props => {
+  const [searchQuery, setSearchQuery] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const [searchedCourses, setSearchedCourses] = useState(EMPTY_COURSE_LIST);
+  const [selectedCourses, setSelectedCourses] = useState(EMPTY_COURSE_LIST);
+  const { schedule } = useSelector((state: AppState) => ({
+    schedule: safelyGetActivePlanScheduleFromState(state),
```

Comment:
I don't think we need to do this safely (the only way to access this modal is through the home screen, which will only render if an active plan exists?)